### ISSUE TITLE
Why it is not possible to serve directly links from playlist.m3u to Plex DVR?

### ISSUE THREAD

**Issue by petrmenzel** [HazCod/muxy#6]
I would like to understand how does it all work. There are: playlist.m3u, (real/virtual) TV tuner, Plex DVR. Why I cannot **serve directly links of streams** from playlist.m3u to Plex DVR? What is **format supported by the Plex** DVR to be playable? Is another purpose of the virtual TV tuner than convert "text" of the playlist.m3u to "text" readable for the Plex DVR? So all should works as **only changing text**?

_Thx you very much for explanation. There is nothing on the web about Plex DVR in deeper form to learn._

**Comment by hazcod** [comment]
Plex does not accept `.TS` files, only the MPEG streams we can decode from it.

**Comment by tombowditch** [comment]
Plex _does_ accept TS files and is accepting the direct URLs I'm shoving at it. Wrote a quick proxy yesterday.

**Comment by hazcod** [comment]
Does it? Hm, then my project is a lot more complex than it needs to be.
I'm looking at your https://github.com/tombowditch/telly

**Comment by tombowditch** [comment]
Yep, wrote that yesterday in an hour or two and works perfectly, there's a few quirks I need to fix but it drops all the overheads and just shoves Plex the .TS URLs from <your IPTV provider of choice>.

**Comment by hazcod** [comment]
@tombowditch The video in your m3u is MPEGTS, right? I'm getting a 'cannot tune to channel' with telly aswel.
Plex: `DEBUG - DVR:Grabber: Operation for Live TV - Session completed with status error (The device does not tune the required channel)`

EDIT: apparently it depends what EPG provider I pick

**Comment by petrmenzel** [comment]
It is strange. How can affect EPG the playback?

But telly does not work for me too. My conclusion is that [muxy](https://github.com/HazCod/muxy) and [telly](https://github.com/tombowditch/telly) do not work but muxy seem to be more stable. The telly TV tuner is telling like the [tvhProxy](https://github.com/jkaberg/tvhProxy) that `Device not found. Ensure your device is powered on and connected to your network.` although it is running.

<img width="1288" alt="telly - command line" src="https://user-images.githubusercontent.com/12900736/35673717-a11076c2-0742-11e8-9374-835cef187bef.PNG">

With the telly the error was `Unable to tune channel (Time out)`.

<img width="526" alt="time out" src="https://user-images.githubusercontent.com/12900736/35673537-25786fba-0742-11e8-8096-b1a356ed7d99.PNG">

Tell me someone please how to recognize that plex is able to play directly stream provided from my ISP? Is it true that all stream playable by VLC should be playable by Plex? My stream looks like this:
```
#EXTINF:0,ČT1
http://media.kuki.cz:8116/ct1hd/stream.m3u8?type=live&sign=XxXxXxXxX&expires=1111111111
```

The `sign` and `expires` is changed every 24 hours. The link perfectly play in the VLC.

Thx you very much. **I think that is important to tell what is the format supported by the Plex to know what is needed to do.**

**Comment by petrmenzel** [comment]
- Plex is probably able to playback **MPEG-2** natively ([What is Plex DVR readable format?](http://forums.plex.tv/discussion/comment/1606596#Comment_1606596)).
- Every IPTV is usually **MPEG2**, MPEG4 stream format. ([IPTV by Chris Woodford. Last updated: January 4, 2018.](http://www.explainthatstuff.com/how-iptv-works.html))
- Stream **.ts** is only type of the MPEG2 so Plex should be able to play .ts to natively. ([What are MPEG-2 TS and MPEG-2 PS?](http://xpt.sourceforge.net/techdocs/media/video/dvdvcd/dv03-VLC/ar01s08.html))

_So important is to discover what kind of stream is mine, yours and in case that it is MPEG2 we can put this stream directly to the Plex DVR._

**Comment by hazcod** [comment]
Can confirm, my streams are MPEG2. So problem must be something else.
Did you try changing EPG provider?

**Comment by tombowditch** [comment]
Hi all, sorry, was at work. 

@HazCod the format of my m3u is something like ...

```
#EXTM3U
#EXTINF:-1, tvg-ID="some name" tvg-name="some name" tvg-logo="some name" group-title="some name",some name
http://my.provider/something/something.ts
```

Running mediainfo on the .ts file shows

```
Format                                   : MPEG-TS
```

Plex accepts it just fine, more than happy to invite you to my Plex Home & Server for you to see ^^

**Comment by tombowditch** [comment]
@petrmenzel 

> The telly TV tuner is telling like the tvhProxy that Device not found. Ensure your device is powered on and connected to your network. although it is running.

Yeah. Looking into this, seems to not work sometimes but sometimes does work. Right now, for example, it's working perfectly:

<img width="364" alt="screen shot 2018-02-01 at 19 24 38" src="https://user-images.githubusercontent.com/5332144/35698630-90351f4a-0785-11e8-9894-28d316578b07.png">

Feel free to open an issue over there and I can look into it more.

**Comment by petrmenzel** [comment]
Now I know my issue and can start to looking for solution.

**My stream is MPEG4.** So Plex does not support play this.

<img width="457" alt="mpeg4" src="https://user-images.githubusercontent.com/12900736/35699160-8a455eec-078f-11e8-954c-f5df5a291a16.PNG">

UPDATE: If the Plex is able to play mpeg4 from .mkv it does not mean that is able to do same thing with mpeg4 stream? ([General Media Support](https://support.plex.tv/articles/203824396-what-media-formats-are-supported/))
